client/consensus/babe/src/lib.rs

```diff
@@ -1627,6 +1680,7 @@ where
 				let prune_and_import = || {
 					prune_finalized(self.client.clone(), &mut epoch_changes)?;
 
+					epoch_changes.sync();
```

Comment:
This is really shitty, I'll try to cleanup this code.

Reply:
For sure we can call this only if required (i.e. if we modified `viable_epoch` content above).
Maybe we can call `epoch_changes.sync()` directly after `viable_epoch.as_mut().start_slot = epoch_start;`

Furthermore maybe we can only modify the fork-tree node corresponding to the altered epoch (instead of remapping the full tree only to change one node).

But I'm sure you're already thinking about it :-)